### ISSUE TITLE
training: confidence head and pattern-attention branch never receive gradients

### ISSUE THREAD

**Issue by mrveiss** [mrveiss/AutoBot-AI#13403]
## Summary

`CompletionModel` defines two branches that the training objective never drives, so both ship with random-initialised weights no matter how long training runs.

Found while fixing `training/completion_trainer_test.py` for #13162 (PR #13402). The test there asserted that a logits-only loss reaches *every* parameter; it cannot, and tracing why surfaced this.

## 1. `confidence_layer` is never trained but is served to callers

`autobot-backend/training/completion_model.py` builds a confidence head:

```
self.confidence_layer = nn.Sequential(
    nn.Linear(hidden_dim * 2, hidden_dim // 2),
    nn.ReLU(),
    nn.Dropout(dropout),
    nn.Linear(hidden_dim // 2, 1),
    nn.Sigmoid(),
)
```

`forward()` computes `confidence` and `predict()` returns it under the `"confidence"` key, so it is part of the model's public output.

But `completion_trainer.train_epoch` takes the loss over logits only:

```
logits, confidence = self.model(input_ids)
loss = self.criterion(logits_flat, targets_flat)
```

`confidence` is bound and discarded. Nothing in the loss depends on `confidence_layer`, so it receives no gradient in any epoch and its output is whatever the initialiser produced. Any consumer treating that number as a calibrated confidence is reading noise.

## 2. The pattern-attention branch is never exercised

`forward()` only applies `self.attention` when `pattern_embeddings` is supplied:

```
if pattern_embeddings is not None:
    attended, _ = self.attention(lstm_out, pattern_embeddings, pattern_embeddings)
    context = attended + lstm_out
else:
    context = lstm_out
```

Both `train_epoch` and `validate` call `self.model(input_ids)` with no pattern embeddings, and no other caller passes them. The `MultiheadAttention` module — described in the class docstring as "Attention mechanism over pattern database" — is allocated, checkpointed and never used.

The branches are correctly wired: PR #13402 adds `test_model_gradient_flow_through_optional_branches`, which drives both and shows gradients reach every parameter. The gap is in the objective and the call sites, not the model.

## Why this was not fixed in PR #13402

Both fixes are product decisions, not defect repairs:

- Training the confidence head needs a target signal to define (correctness of the top-1 prediction? calibration against held-out accuracy?) and a loss-weighting choice.
- Driving the attention branch needs the pattern database wired into the data pipeline so `pattern_embeddings` can be produced per batch.

Either changes what the model learns, so it needs an owner decision rather than a scattered-test-fix PR.

## Proposed work

- [ ] Decide the confidence target and add the term to the training objective, or remove the head from the public `predict()` output until it is trained
- [ ] Wire the pattern database into `prepare_data` so `pattern_embeddings` reach `forward()`, or record why the branch stays dormant
- [ ] Extend the trainer tests to assert the chosen objective reaches every parameter

Related: #13162, PR #13402
